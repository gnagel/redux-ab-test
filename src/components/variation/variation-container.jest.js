import React from 'react'; // eslint-disable-line no-unused-vars
import Immutable from 'immutable';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Variation, { mapStateToProps } from './variation';
import reduxAbTest, { initialState } from '../../module';


describe('(Container) Variation', () => {
  describe('mapStateToProps', () => {
    it('exists', () => {
      expect(mapStateToProps).not.toBeUndefined;
    });
    it('has correct keys', () => {
      expect(Object.keys(mapStateToProps({reduxAbTest: 'test-state'}))).toEqual(['reduxAbTest']);
    });
    it('has correct state', () => {
      expect(mapStateToProps({reduxAbTest: 'test-state'})).toEqual({reduxAbTest: 'test-state'});
    });
  });

  describe('component', () => {
    let component;
    let tree;
    beforeEach(() => {
      const props = {
        name:       'Variation B',
        experiment: Immutable.fromJS({
          name:       'Test-experimentName',
          variations: [
            { name: 'Original',  weight: 10000 },
            { name: 'Variation B', weight: 0 },
          ],
        }),
        children: 'Test Variation B',
      };
      const state = {
        reduxAbTest: initialState.merge({
          experiments: [
            {
              name:       'Test-experimentName',
              variations: [
                { name: 'Original', weight: 10000 },
                { name: 'Variation B', weight: 0 },
              ],
            },
          ],
        }),
      };
      component = renderer.create(
        <Provider store={createStore(reduxAbTest, state)}>
          <Variation {...props} />
        </Provider>
      );
      tree = component.toJSON();
    });

    it('variation has expected props', () => {
      expect(tree).toMatchSnapshot();
      expect(tree.type).toEqual('span');
      expect(tree.props['data-experiment-id']).toBeUndefined;
      expect(tree.props['data-experiment-name']).toEqual('Test-experimentName');
      expect(tree.props['data-variation-id']).toBeUndefined;
      expect(tree.props['data-variation-name']).toEqual('Variation B');
      expect(tree.children).toEqual(['Test Variation B']);
    });
  });
});
