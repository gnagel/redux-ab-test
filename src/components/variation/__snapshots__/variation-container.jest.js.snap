// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) Variation component variation has expected props 1`] = `
<span
  data-experiment-id={undefined}
  data-experiment-name="Test-experimentName"
  data-variation-id={undefined}
  data-variation-name="Variation B"
>
  Test Variation B
</span>
`;
